File: business-web/src/utils/formatters.ts
/**
 * AVENZO Business Web — Utility Formatters
 * Centralized formatting helpers for Currency (INR ₹), Dates, and Expiry Badges.
 */

/**
 * Formats monetary amounts in INR (₹).
 * Centralized formatter per Phase 3 architectural requirement (Section 13).
 */
export function formatCurrency(amount: number | string | null | undefined): string {
  if (amount === null || amount === undefined) {
    return '₹0.00';
  }
  const numericVal = typeof amount === 'string' ? parseFloat(amount) : amount;
  if (isNaN(numericVal)) {
    return '₹0.00';
  }
  return new Intl.NumberFormat('en-IN', {
    style: 'currency',
    currency: 'INR',
    maximumFractionDigits: 2,
    minimumFractionDigits: 2,
  }).format(numericVal);
}

/**
 * Formats ISO date strings to human-readable format (e.g. 15 Aug 2026).
 */
export function formatDate(dateString: string | null | undefined): string {
  if (!dateString) return 'N/A';
  const d = new Date(dateString);
  if (isNaN(d.getTime())) return 'N/A';
  return new Intl.DateTimeFormat('en-IN', {
    day: 'numeric',
    month: 'short',
    year: 'numeric',
  }).format(d);
}

export interface ExpiryStatusBadgeProps {
  status: string;
  label: string;
  variant: 'success' | 'warning' | 'danger' | 'expired' | 'neutral';
  icon: string;
}

/**
 * Returns accessible badge properties for Expiry Status classification.
 */
export function getExpiryBadgeProps(status: string): ExpiryStatusBadgeProps {
  switch (status?.toUpperCase()) {
    case 'SAFE':
      return {
        status: 'SAFE',
        label: 'SAFE (>30 days)',
        variant: 'success',
        icon: '✓',
      };
    case 'EXPIRING_SOON':
      return {
        status: 'EXPIRING_SOON',
        label: 'EXPIRING SOON (8-30 days)',
        variant: 'warning',
        icon: '⚠',
      };
    case 'CRITICAL':
      return {
        status: 'CRITICAL',
        label: 'CRITICAL (0-7 days)',
        variant: 'danger',
        icon: '🔥',
      };
    case 'EXPIRED':
      return {
        status: 'EXPIRED',
        label: 'EXPIRED (<0 days)',
        variant: 'expired',
        icon: '⛔',
      };
    default:
      return {
        status: 'N/A',
        label: 'NON-EXPIRY',
        variant: 'neutral',
        icon: '•',
      };
  }
}

File: consumer-app/lib/core/utils/formatters.dart
import 'package:intl/intl.dart';

/// Formatting utilities for dates, currency, and DTE (Days to Expiry).
class Formatters {
  Formatters._();

  static final _currencyFormat = NumberFormat.currency(
    symbol: '₹',
    locale: 'en_IN',
    decimalDigits: 2,
  );

  /// Format numeric price into INR standard (₹1,250.00)
  static String currency(dynamic amount) {
    if (amount == null) return '₹0.00';
    final numValue = amount is String ? double.tryParse(amount) ?? 0.0 : amount as num;
    return _currencyFormat.format(numValue);
  }

  /// Format date to readable string (e.g., "18 Aug 2026")
  static String formatDate(DateTime? date) {
    if (date == null) return 'N/A';
    return DateFormat('d MMM yyyy').format(date);
  }

  /// Format DTE (Days to Expiry) badge label
  static String formatDTE(DateTime? expiryDate) {
    if (expiryDate == null) return 'No Expiry';
    final now = DateTime.now();
    final today = DateTime(now.year, now.month, now.day);
    final expiry = DateTime(expiryDate.year, expiryDate.month, expiryDate.day);
    final days = expiry.difference(today).inDays;

    if (days < 0) {
      return 'Expired ${days.abs()}d ago';
    } else if (days == 0) {
      return 'Expires Today';
    } else if (days == 1) {
      return 'Expires Tomorrow';
    } else {
      return '$days days left';
    }
  }
}
